### ISSUE TITLE
TestBigMatrix fails on mips64el and ppc64el architectures

### ISSUE THREAD

**Issue by merkys** [simbody/simbody#751]
Simbody 3.7 fails TestBigMatrix when built on Debian on mips64el and ppc64el architectures:
```
test 12
        Start  12: TestBigMatrix

12: Test command: obj-mips64el-linux-gnuabi64/TestBigMatrix
12: Working Directory: obj-mips64el-linux-gnuabi64/SimTKcommon/tests
12: Test timeout computed to be: 1500
12: f(v)=~[1,2,3]
12: exception: SimTK Exception thrown at TestBigMatrix.cpp:100:
12:   Internal bug detected: Test condition failed.
12:   (Assertion '(X*vs == -(X*-vs))' failed).
12:   Please file an Issue at https://github.com/simbody/simbody/issues.
12:   Include the above information and anything else needed to reproduce the problem.
  9/108 Test  #12: TestBigMatrix ..................................***Failed    0.02 sec
f(v)=~[1,2,3]
exception: SimTK Exception thrown at TestBigMatrix.cpp:100:
  Internal bug detected: Test condition failed.
  (Assertion '(X*vs == -(X*-vs))' failed).
  Please file an Issue at https://github.com/simbody/simbody/issues.
  Include the above information and anything else needed to reproduce the problem.

```
Full build logs: [mips64el](https://buildd.debian.org/status/fetch.php?pkg=simbody&arch=mips64el&ver=3.7%2Bdfsg-2&stamp=1669708062&raw=0) and [ppc64el](https://buildd.debian.org/status/fetch.php?pkg=simbody&arch=ppc64el&ver=3.7%2Bdfsg-2&stamp=1669706492&raw=0). Log files are ~100 MB in size.

**Comment by sherm1** [comment]
Is this something that was working previously and stopped working? If so, any idea what broke it? (Simbody code change, build compiler change, etc?)

**Comment by merkys** [comment]
Yes, TestBigMatrix used to pass on mips64el and ppc64el architectures. Doing some log bisecting I find the following:

| Simbody version | C/CXX compiler version | TestBigMatrix |
| --- | --- | --- |
| 3.6.1 | 8.3.0 | Passed |
| 3.6.1 | 12.2.0 | Failed |

Thus it seems that the problem for these architectures is the update of C/CXX compiler from 8.3.0 to 12.2.0.

**Comment by sherm1** [comment]
I'm not sure what to do with this. The problem appears to be an optimizer bug in the mips and ppc backends for the newer compiler, but I don't have access to those so can't confirm that compiling with optimization off allows the test to pass. Do you have an idea for how to address this problem?

**Comment by merkys** [comment]
I can try disabling optimization for these architectures just to see if that helps. How can I do that?

**Comment by sherm1** [comment]
@merkys -- sorry for the delayed response!

As an experiment, please try modifying line [CMakeLists.txt:400](https://github.com/simbody/simbody/blob/master/CMakeLists.txt#L400) at the top level to read
`set(GCC_OPT_DISABLE -O1)` or `-O0` if it still fails at O1.
Then regenerate with CMake and rebuild to see of all tests pass.

If that works we can PR a conditional on the architecture.

**Comment by merkys** [comment]
Just checked, neither `-O1`, nor `-O0` makes TestBigMatrix pass, alas.

**Comment by sherm1** [comment]
Rats! That's surprising. Thanks for trying. Is the compiler you're using gcc or clang?
